gst/gst_debug.go

```diff
@@ -164,3 +187,23 @@ func (d *DebugCategory) LogTrace(message string, obj ...*Object) {
 func (d *DebugCategory) LogMemDump(message string, obj ...*Object) {
 	d.logDepth(LevelMemDump, message, 2, getLogObj(obj...))
 }
+
+type LogFunction func(
+	level DebugLevel,
+	file string,
+	function string,
+	line int,
+	object *glib.Object,
+	message string,
+)
+
+var customLogFunction LogFunction
+
+func SetLogFunction(f LogFunction) {
+	if f == nil {
+		C.cgoUnsetLogFunction()
+	} else if customLogFunction == nil {
+		C.cgoSetLogFunction()
+	}
```

Comment:
If we want to be exact here, and be certain that all goroutines stated before `SetLogFunction` is called see the updated value of customLogFunction, we need some kind of memory barrier here, the easiest being a mutex...